plugins/colorize/colorize.plugin.zsh

```diff
@@ -61,22 +67,41 @@ colorize_via_pygmentize() {
     done
 }
 
-colorize_via_pygmentize_less() (
-    # this function is a subshell so tmp_files can be shared to cleanup function
-    declare -a tmp_files
+# The less option 'F - Forward forever; like "tail -f".' will not work in this implementation
+# caused by the lack of the ability to follow the file within pygmentize.
+colorize_less() {
+    if ! colorize_check_requirements; then
+        return 1
+    fi
 
-    cleanup () {
-        [[ ${#tmp_files} -gt 0 ]] && rm -f "${tmp_files[@]}"
-        exit
-    }
-    trap 'cleanup' EXIT HUP TERM INT
+    # LESS="-R $LESS" enables raw ANSI colors, while maintain already set options.
+    local LESS="-R $LESS"
 
-    while (( $# != 0 )); do     #TODO: filter out less opts
-        tmp_file="$(mktemp -t "tmp.colorize.XXXX.$(sed 's/\//./g' <<< "$1")")"
-        tmp_files+=("$tmp_file")
-        colorize_via_pygmentize "$1" > "$tmp_file"
-        shift 1
-    done
+    # LESSOPEN="|zsh -c 'source \"$ZSH_COLORIZE_PLUGIN_PATH\"; colorize_cat %s 2> /dev/null'"
+    #   This variable tells less to pipe every file through the specified command (see the man of less INPUT PREPROCESSOR).
+    #   'zsh -ic "colorize_cat %s 2> /dev/null"' would not work for huge files like the ~/.zsh_history.
+    #   For such files the tty of the preprocessor will be supended.
+    #   Therefore we must source this file to make colorize_cat available in the preprocessor without the interactive mode.
+    #   2> /dev/null will suppress the error for large files 'broken pipe' of the python script pygmentize,
+    #   which will show up if less has not fully "loaded the file" (e.g. when not scrolled to the bottom)
+    #   while already the next file will be displayed.
+    local LESSOPEN="|zsh -c 'source \"$ZSH_COLORIZE_PLUGIN_PATH\"; colorize_cat %s 2> /dev/null'"
 
-    less -f "${tmp_files[@]}"
-)
+    # LESSCLOSE will be set to prevent any errors by executing an user script which assumes that his LESSOPEN has been executed.
+    local LESSCLOSE=""
+
+    _cless() {
+      LESS="$LESS" LESSOPEN="$LESSOPEN" LESSCLOSE="$LESSCLOSE" less "$@"
```

Comment:
This `LESSCLOSE="$LESSCLOSE"` is the same as `LESSCLOSE=""`, without having to redefine LESSCLOSE as a local variable. I would also put the variables inside the `_cless` function as well because it's where they're used.

Reply:
Your right, the variables should be inside the function. But I created the variable LESSCLOSE on purpose, so it is consistent to the others and can be directly commented. If you wish, I'll remove it.

Reply:
I guess we can keep it for consistency, yeah.